main.tf

```diff
@@ -101,45 +122,42 @@ resource "ibm_is_vpc" "vpc" {
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_id" {
   count = (var.enable_hub == false && var.enable_hub_vpc_id) ? 1 : 0
 
-  name   = "${var.prefix}-dns-binding"
+  # Use var.dns_binding_name if not null, otherwise, use var.prefix or var.name.
+  name   = coalesce(var.dns_binding_name, "${var.prefix != null ? var.prefix : var.name}-dns-binding")
   vpc_id = local.vpc_id # Source VPC
   vpc {
     id = var.hub_vpc_id # Target VPC ID
   }
 }
 
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_crn" {
-  count  = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
-  name   = "${var.prefix}-dns-binding"
+  count = (var.enable_hub == false && var.enable_hub_vpc_crn) ? 1 : 0
+
+  # Use var.dns_binding_name if not null, otherwise, use var.prefix or var.name.
+  name   = coalesce(var.dns_binding_name, "${var.prefix != null ? var.prefix : var.name}-dns-binding")
   vpc_id = local.vpc_id # Source VPC
   vpc {
     crn = var.hub_vpc_crn # Target VPC CRN
   }
 }
 
-data "ibm_is_vpc" "vpc" {
-  count      = var.create_vpc == false ? 1 : 0
-  identifier = var.existing_vpc_id
-}
-
-locals {
-  vpc_id   = var.create_vpc ? ibm_is_vpc.vpc[0].id : var.existing_vpc_id
-  vpc_data = var.create_vpc ? ibm_is_vpc.vpc[0] : data.ibm_is_vpc.vpc[0]
-}
-
 # Configure custom resolver on the hub vpc
 resource "ibm_resource_instance" "dns_instance_hub" {
-  count             = var.enable_hub && !var.skip_custom_resolver_hub_creation && !var.use_existing_dns_instance ? 1 : 0
-  name              = "${var.prefix}-dns-instance"
+  count = var.enable_hub && !var.skip_custom_resolver_hub_creation && !var.use_existing_dns_instance ? 1 : 0
+
+  # Use var.dns_instance_name if not null, otherwise, use var.prefix or var.name.
+  name              = coalesce(var.dns_instance_name, "${var.prefix != null ? var.prefix : var.name}-dns-instance")
```

Comment:
If the consumer passed a value for `prefix` and for `name` (and no value for `dns_instance_name `), then I think we will want the dns instance name name to be called `${var.prefix}-${var.name}-dns-instance`. Currently logic does not take that into consideration.